Test/NilException.m

```diff
@@ -0,0 +1,17 @@
+#include "Test.h"
+
+int main(void)
+{
+	BOOL caught_exception = NO;
+	@try
+	{
+		@throw(nil);
+	}
+	@catch (id x)
+	{
+		assert(nil == x);
+		caught_exception = YES;
+	}
```

Comment:
What's the correct behaviour for, throwing `nil` and catching  (for example)`NSObject *`?  Should `nil` also be caught by `void*` catches in Objective-C++ (this case used to crash the Apple runtime, not sure if they've fixed it now)?